src/include/simeng/branchpredictors/PerceptronPredictor.hh

```diff
@@ -59,16 +69,28 @@ class PerceptronPredictor : public BranchPredictor {
    * in Jiminez and Lin */
   std::vector<std::pair<std::vector<int8_t>, uint64_t>> btb_;
 
-  /** The previous hashed index for an address. */
-  std::map<uint64_t, uint64_t> btbHistory_;
+  /** Fetch Target Queue containing the dot product of the perceptron and the
+   * global history; and the global history, both at the time of prediction,
+   * for each of the branches that are currently unresolved. */
+  std::deque<std::pair<int64_t, uint64_t>> ftq_;
```

Comment:
first "and" -> "with". Even better if changed to what this value means e.g. the predicted direction of the branch.

Could go to: "FTQ containing predicted direction and the global history at the time of the prediction for all branches that are currently [unresolved/in-flight]"

Reply:
I think "and" is correct here (at least its what the MIT maths department use: https://www.mit.edu/~hlb/StantonGrant/18.02/details/tex/lec1snip2-dotprod.pdf).  I've added to the confidence the significance of the dot product.

Reply:
"And" on its own is fine. I just thought it would read better with the change considering the second "and" and the rest of the sentence. The second half of the comment is great